Client/Frontend/Home/ActivityStreamSupplementary.swift

```diff
@@ -0,0 +1,310 @@
+        titleLabel.backgroundColor = TopSiteCellUX.TitleBackgroundColor
+        contentView.addSubview(titleLabel)
+
+        let heightInset = Int(frame.height * TopSiteCellUX.TitleInsetPercent)
+        titleLabel.snp_makeConstraints { make in
+            make.edges.equalTo(self).inset(UIEdgeInsetsMake(CGFloat(heightInset), 0, 0, 0))
+        }
+
+        imageView = UIImageView()
+        imageView.layer.masksToBounds = true
+        contentView.addSubview(imageView)
+        imageView.snp_makeConstraints { make in
+            make.height.equalTo(self.frame.height/2)
+            make.width.equalTo(self.frame.height/2)
+
+            // Add an offset to the image to make it appear centered with the titleLabel
+            let offset = Int(self.frame.height) - heightInset
+            make.centerX.equalTo(self.snp_centerX)
+            make.centerY.equalTo(self.snp_centerY).offset(CGFloat(-offset/2))
+        }
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func prepareForReuse() {
+        backgroundColor = UIColor.whiteColor()
+        imageView.image = nil
+        titleLabel.text = ""
+    }
+
+    func setImageWithURL(url: NSURL) {
+        imageView.sd_setImageWithURL(url) { (img, err, type, url) -> Void in
+            guard let img = img else {
+                // No favicon found. Do something!
+                return
+            }
+
+            img.getColors(CGSize(width: 50, height:50)) { colors in
+                //In cases where the background is white. Force the background color to a different color
+                var bgColor: UIColor
+                if colors.backgroundColor.isWhite {
+                    let colorArr = [colors.detailColor, colors.primaryColor].filter {return !$0.isWhite}
+                    if colorArr.isEmpty {
+                        ///need an array of default colors
+                        bgColor = UIColor.greenColor()
+                    }
+                    else {
+                        bgColor = colorArr[0]
+                    }
+                }
+                else {
+                    bgColor = colors.backgroundColor
+                }
+                contentView.backgroundColor = bgColor
+
+            }
+        }
+    }
+
+    func configureWithTopSiteItem(site: TopSiteItem) {
+        titleLabel.text = site.urlTitle
+        setImageWithURL(site.faviconURL)
+    }
+
+    func makeCellBlank() {
+        backgroundColor = UIColor.whiteColor()
+        titleLabel.backgroundColor = UIColor.whiteColor()
+        imageView.backgroundColor = UIColor.whiteColor()
+        contentView.backgroundColor = UIColor.whiteColor()
+    }
+
+}
+
+class ASHorizontalScrollCell: UITableViewCell {
+    private var collectionView: UICollectionView!
+    private var pageControl: UIPageControl!
+    private var headerView: ASHeaderView!
+
+    var headerTitle: String = "" {
```

Comment:
Instead of letting external classes mess with headerView. I created a headerTitle that will update the headerView's title